tests/infer/test_abstract_infer.py

```diff
@@ -44,3 +47,30 @@ def nested():
         nested()
 
     assert len(tp.trace.nodes) == 0
+
+
+@pytest.mark.init(rng_seed=3)
```

Comment:
How variable are the results with the default seed? Is it possible to relax the thresholds instead? A consequence of adding as many tests as we have now, is that we need to be careful in ensuring that they can run robustly under different environments - CUDA, platform etc, otherwise maintaining them will be challenge.

Reply:
@neerajprad As long as AutoDelta is able to find the model's MAP points. Then we can get the correct WAIC. However, depending on seed (hence on initial point of AutoDelta), it took us longer to converge. I'll find good lr and num_steps which ensures converging at with different seeds though.

Reply:
@neerajprad Correction: with this lr, model converges for different seeds. But with different seeds, we have different posterior samples. That causes different results in waic. Sorry for confusion. :(